sympy/integrals/transforms.py

```diff
@@ -1836,7 +1836,7 @@ def _try_directly(self, **hints):
         return fn, LT
 
 
-def laplace_transform(f, t, s, legacy_matrix=True, **hints):
+def laplace_transform(f, t, s, diff_subs=dict(), legacy_matrix=True, **hints):
```

Comment:
I'm not sure `diff_subs` is such a good name for this parameter. Maybe `function_map` or something is better? Differentiation need not be involved:
```python
In [1]: t, s = symbols('t, s')

In [2]: x, X = symbols('x, X', cls=Function)

In [4]: laplace_transform(x(t), t, s, {x(t):(X(s),[])})
Out[4]: X(s)
```
Also I think this parameter should be keyword-only rather than positional.

Reply:
Shouldm't legacy_matrix then also be a keyword-only argument?

Reply:
Yes, but we should really just get rid of legacy_matrix.